### ISSUE TITLE
Edge cases for row-major ukernels not handled quite right

### ISSUE THREAD

**Issue by devinamatthews** [flame/blis#78]
For `gemm` ukernels which prefer contiguous rows, the temporary buffer for edge cases (when `m != mr` or `n != nr`) is treated as column-major (i.e. `rs_c = 1` and `cs_c = mr`). If the general-stride pathway in the ukernel is very slow then this might have a performance impact for small to medium size matrices.

The ukernel wrapper should treat the temporary buffer as row-major (`rs_c = nr` and `cs_c = 1`) in this case.

**Comment by fgvanzee** [comment]
Thanks for catching this, @devinamatthews. I agree that there is room for performance improvement in those cases.

**Comment by fgvanzee** [comment]
@devinamatthews Towards this fix, do you agree with the following patch to each of the macro-kernels?

``` 
-       const inc_t     rs_ct      = 1; \
-       const inc_t     cs_ct      = MR; \
+       const inc_t     rs_ct      = ( bli_is_col_stored( rs_c, cs_c ) ? 1 : NR ); \
+       const inc_t     cs_ct      = ( bli_is_col_stored( rs_c, cs_c ) ? MR : 1 ); \
```
(Turns out this is actually *necessary* for some of the complex domain stuff I'm doing now.)

**Comment by devinamatthews** [comment]
Is there a reason not to query the kernel row major property directly?

**Comment by fgvanzee** [comment]
@devinamatthews I suppose that would be more direct. The fix I proposed above assumes that the higher-level code has already oriented the strides of C to match the storage preference of the microkernel, as accomplished by the following snippet from `bli_gemm_front.c`:

```
    if ( bli_cntx_l3_ukr_dislikes_storage_of( &c_local, BLIS_GEMM_UKR, cntx ) )
    {
        bli_obj_swap( a_local, b_local );

        bli_obj_induce_trans( a_local );
        bli_obj_induce_trans( b_local );
        bli_obj_induce_trans( c_local );
    }
```

**Comment by devinamatthews** [comment]
I guess either works.

**Comment by fgvanzee** [comment]
This issue should now be fixed in 618f433.
